djangoproject/middleware.py

```diff
@@ -10,3 +14,23 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Access-Control-Allow-Origin'] = '*'
         return response
+
+
+class ExcludeDomainsLocaleMiddleware(LocaleMiddleware):
+    """
+    Locale middleware that lets us exclude requests on certain domains (e.g.,
+    docs.djangoproject.com) from being processed by LocaleMiddleware.
+    """
+
+    def _is_domain_excluded(self, request):
+        excluded_domains = getattr(settings, 'LOCALE_MIDDLEWARE_EXCLUDED_DOMAINS', [])
+        return 'HTTP_HOST' in request.META and request.META['HTTP_HOST'] in excluded_domains
```

Comment:
You want [`request.get_host()` here to make sure proxying headers are taken into consideration](https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.get_host)

```suggestion
        return request.get_host() in self.excluded_domains
```